### PR TITLE
Make vault_disable_mlock a role parameter defaulted to false

### PR DESCRIPTION
**Pull request by nununo**
Proposal for implementing issue #287 which I posted earlier today.

### CODE REVIEW

File: templates/vault_main_configuration.hcl.j2 [hunk 1]
```diff
@@ -87,6 +87,9 @@ listener "tcp" {
   {% include vault_backend_raft with context -%}
 {% endif %}
 
+{% if vault_disable_mlock -%}
```
Comment (bbaassssiiee) [inline templates/vault_main_configuration.hcl.j2]: I'd expect `if vault_disable_mlock is defined`

Comment (nununo) [reply templates/vault_main_configuration.hcl.j2]: Oh my! I believe you're right.

Comment (nununo) [reply templates/vault_main_configuration.hcl.j2]: I just submitted a correction.

**Comment by nununo** [comment]
The template `vault_main_configuration.hcl.j2` seems to have changed substantially since I submitted this PR. Hence the conflicts. I'm not sure the problem that this PR intends to solve is gone but, as it is, this PR is no longer relevant so I will close it.

**Pull request closed without merging**